backend/src/routes/api/storage/index.ts

```diff
@@ -0,0 +1,50 @@
+import { FastifyInstance } from 'fastify';
+import { createMinioClient, getObjectStream } from './storageUtils';
+import { getDashboardConfig } from '../../../utils/resourceUtils';
+
+export default async (fastify: FastifyInstance): Promise<void> => {
+  const dashConfig = getDashboardConfig();
+  if (!dashConfig.spec.dashboardConfig.disableS3Endpoint) {
```

Comment:
It's a stupid config -- sadly, this won't work.

disableS3Endpoint equals:
* `true` (flag off) => !true => false ✅ 
* `false` (flag on) => !false => true ✅ 
* `undefined` (flag not set) => !undefined => true ❌  

Explicitly check to see if it's `=== false`